internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -188,6 +204,110 @@ func (c *cmd) GetCommandName() string {
 	return CommandName
 }
 
+// createEdgeXACLTokenRoles creates secret store roles that can be used for genearting registry tokens
+// via Consul secret engine API /consul/creds/[role_name] later on for all EdgeX microservices
+func (c *cmd) createEdgeXACLTokenRoles(secretstoreToken string, rolePolicy Policy) error {
+	roleNames := c.configuration.StageGate.Registry.ACL.GetACLRoleNames()
+	if len(roleNames) == 0 {
+		c.loggingClient.Warn("found no ACL role names defined in configuration, skip create ACL roles")
+	}
+
+	for _, roleName := range roleNames {
+		// create roles based on the service keys as the role names
+		// in phase 2, we are using the same policy rule for all services
+		edgexACLTokenRole := NewRegistryRole(roleName, ClientType, []Policy{
+			rolePolicy,
+		}, false)
+
+		// fail all if any one of the role creation failed
+		if err := c.createRole(secretstoreToken, edgexACLTokenRole); err != nil {
+			return fmt.Errorf("failed to create edgex role: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *cmd) createAndSaveManagementToken(bootstrapACLTokenID string) error {
+	mgmtPolicyName := "edgex-management-policy"
+	mgmtPolciyRules := `
+	# EdgeX management policy for management token in HCL format
+	acl = "read"
+	agent_prefix "" {
+		policy = "write"
+	}
+	key_prefix "" {
+		policy = "write"
+	}
+	node_prefix "" {
+		policy = "write"
+	}
+	operator = "write"
+	service_prefix "" {
+		policy = "write"
+	}
+	session_prefix "" {
+		policy = "write"
+	}
+	`
```

Comment:
Will we ever need to changes this? If so it should be moved to configuration.

Reply:
once it is settled, it won't be changed any more

Reply:
ok, for now.